taxjar/taxjar-magento2-extension
# Check if transaction sync is disabled before sync

### Context
This PR limits syncing transactions (aka backfilling) when a specific website/store has disabled transaction sync.  It also restricts the transactions that are backfilled to the current scope config in the admin.

### Description
This PR moves the checks to see if a transaction can be backfilled to the `isSyncable` method.  It checks the store associated with the specific order confirm transaction sync is enabled for that store.  

### Performance
This PR should improve performance as it reduces the number of transactions synced in certain situations.  

### Testing
1. Ensure the "Transaction Sync" admin configuration (Stores > Configuration > Sales > Tax > TaxJar) is set to "Yes"
![Screen Shot 2020-03-19 at 1 50 36 PM](https://user-images.githubusercontent.com/44789510/77110151-b6e5d200-69ea-11ea-9df5-4c5a4543a44e.png)
2. Change the admin scope to a website or store view.  
![Screen Shot 2020-03-19 at 1 48 58 PM](https://user-images.githubusercontent.com/44789510/77110190-c5cc8480-69ea-11ea-8cff-83474f7d23a8.png)
3.  Change "Transaction Sync" from "Yes" to "No"
4. Change the scope back to "Default" and confirm that associated transactions don't sync.  (ex:  #2000000001)
![Screen Shot 2020-03-19 at 1 48 52 PM](https://user-images.githubusercontent.com/44789510/77110271-ec8abb00-69ea-11ea-9134-caf3cd1058d9.png)




#### Versions
